bokehjs/src/lib/models/widgets/tables/cell_formatters.ts

```diff
@@ -56,37 +59,96 @@ export class StringFormatter extends CellFormatter {
+      text_align: [ p.TextAlignSpec, {value: "left"} ],
+      text_color: [ p.ColorSpec, null ],
+      background_color: [ p.ColorSpec, null ],
       nan_format: [ String, "-"],
     }))
   }
 
   override doFormat(_row: any, _cell: any, value: any, _columnDef: any, _dataContext: any): string {
-    const {font_style, text_align, text_color} = this
+    const {font_style, text_align, text_color, background_color} = this
 
     const text = div(value == null ? "" : `${value}`)
-    switch (font_style) {
+
+    // Font style
+    let resolved_font_style
+    if (isValue(font_style)) {
+      resolved_font_style = font_style.value
+    } else if (isField(font_style)) {
+      resolved_font_style = _dataContext[font_style.field]
+    } else if (isExpr(font_style)) {
+      // TODO
+    } else {
+      unreachable()
+    }
+
+    switch (resolved_font_style) {
       case "normal":
-        break
-      case "bold":
-        text.style.fontWeight = "bold"
+        // nothing to do
         break
       case "italic":
         text.style.fontStyle = "italic"
         break
-      case "bold italic":
+      case "bold":
         text.style.fontWeight = "bold"
+        break
+      case "bold italic":
         text.style.fontStyle = "italic"
+        text.style.fontWeight = "bold"
         break
     }
 
-    text.style.textAlign = text_align
+    // Text align
+    if (isValue(text_align)) {
+      text.style.textAlign = text_align.value
+    } else if (isField(text_align)) {
+      text.style.textAlign = _dataContext[text_align.field]
+    } else if (isExpr(text_align)) {
+      // TODO
+    } else {
+      unreachable()
+    }
+
+    // Text color
+    // Handle the most common case first : isValue and value == null
+    if (isValue(text_color)) {
+      if (text_color.value != null) {
+        text.style.color = color2css(text_color.value)
+      }
+    } else if (isField(text_color)) {
+      if (text_color.transform != null && text_color.transform instanceof ColorMapper) {
+        const rgbaArray = text_color.transform.rgba_mapper.v_compute([_dataContext[text_color.field]])
+        const [r, g, b, a] = rgbaArray
+        text.style.color = rgba2css([r, g, b, a])
+      } else {
+        text.style.color = _dataContext[text_color.field]
+      }
+    } else if (isExpr(text_color)) {
+      // TODO
+    } else {
+      unreachable()
+    }
 
-    if (text_color != null) {
-      text.style.color = color2css(text_color)
+    // Background color
+    if (isValue(background_color)) {
+      if (background_color.value != null) {
+        text.style.backgroundColor = color2css(background_color.value)
+      }
+    } else if (isField(background_color)) {
+      if (background_color.transform != null && background_color.transform instanceof ColorMapper) {
+        const rgbaArray = background_color.transform.rgba_mapper.v_compute([_dataContext[background_color.field]])
+        const [r, g, b, a] = rgbaArray
```

Comment:
This is a bit confusing at the first glance, but given that `v_compute` returns a flat array (technically an ndarray), then this is indeed correct. Sadly there is no scalar variant in `ColorMapper`, so we will have to keep this as-is.